xcube/cli/_gen2/callback.py

```diff
@@ -30,17 +35,48 @@
 
 
 class CallbackApiProgressObserver(ProgressObserver):
-    def __init__(self, callback_api_url: str):
-        self.callback_api_url = callback_api_url
+    def __init__(self, callback_api_cfg: Request):
```

Comment:
```suggestion
    def __init__(self, callback_config: CallbackConfig):
```

Following least knowledge principle: `Request` is too much; `CallbackConfig` should be enough.